pkg/controller/bundle/bundle_unpacker.go

```diff
@@ -165,26 +185,62 @@ func (c *ConfigMapUnpacker) job(cmRef *corev1.ObjectReference, bundlePath string
 	job.SetName(cmRef.Name)
 	job.SetOwnerReferences([]metav1.OwnerReference{ownerRef(cmRef)})
 
+	// By default the BackoffLimit is set to 6 which with exponential backoff 10s + 20s + 40s ...
+	// translates to ~10m of waiting time.
+	// We want to fail faster than that when we have repeated failures from the bundle unpack pod
+	// so we set it to 3 which is ~1m of waiting time
+	// See: https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-backoff-failure-policy
+	// TODO (haseeb): Should this be configurable as well?
```

Comment:
It might help to adopt an idiom like: "present the smallest possible surface to users" -- i.e. if we're unsure of whether a knob is necessary, don't include it.

Anything we expose becomes an API; the less we add to it, the easier it is to support and extend.